server/config.go

```diff
@@ -84,16 +68,9 @@ func ParseConfig(bytes []byte) (*Config, error) {
 	}
 	c.Server.SetValuesFromEnv(envPrefix)
 
-	if v, ok := os.LookupEnv(envPrefix + "PUSH_RESTRICTION_USER_TOKEN"); ok {
-		c.Options.PushRestrictionUserToken = v
-	}
-
-	if c.Options.SharedRepository == "" {
-		c.Options.SharedRepository = DefaultSharedRepository
-	}
-	if c.Options.SharedConfigurationPath == "" {
-		c.Options.SharedConfigurationPath = DefaultSharedConfigurationPath
-	}
+	c.Options.SetValuesFromEnv("BULLDOZER_OPTIONS_")
+	// BULLDOZER_ENV_PREFIX_ will override the BULLDOZER_OPTIONS_ values.
+	c.Options.SetValuesFromEnv("BULLDOZER_ENV_PREFIX_")
```

Comment:
`BULLDOZER_ENV_PREFIX` is meant to be a replacement for the `BULLDOZER_` part of the environment variable. After that, the convention I've tried to use for additional app configuration is that the variable name should follow the path in the YAML file. Specifically, I think these lines should be:

```suggestion
	c.Options.SetValuesFromEnv(envPrefix + "OPTIONS_")
```

By default, I would use e.g. `BULLDOZER_OPTIONS_APP_NAME`. If I also set `BULLDOZER_ENV_PREFIX=TEST_`, then I should use `TEST_OPTIONS_APP_NAME`.

Reply:
So that means the following:

| YAML name                           | ENV name                                      | ENV override name                                        |
|:------------------------------------|:----------------------------------------------|:---------------------------------------------------------|
| options.configuration_path          | BULLDOZER_OPTIONS_CONFIGURATION_PATH          | BULLDOZER_ENV_PREFIX_OPTIONS_CONFIGURATION_PATH          |
| options,app_name                    | BULLDOZER_OPTIONS_APP_NAME                    | BULLDOZER_ENV_PREFIX_OPTIONS_APP_NAME                    |
| options,push_restriction_user_token | BULLDOZER_OPTIONS_PUSH_RESTRICTION_USER_TOKEN | BULLDOZER_ENV_PREFIX_OPTIONS_PUSH_RESTRICTION_USER_TOKEN |
| options,shared_repository           | BULLDOZER_OPTIONS_SHARED_REPOSITORY           | BULLDOZER_ENV_PREFIX_OPTIONS_SHARED_REPOSITORY           |
| options,shared_configuration_path   | BULLDOZER_OPTIONS_SHARED_CONFIGURATION_PATH   | BULLDOZER_ENV_PREFIX_OPTIONS_SHARED_CONFIGURATION_PATH   |
| options,disable_update_feature      | BULLDOZER_OPTIONS_DISABLE_UPDATE_FEATURE      | BULLDOZER_ENV_PREFIX_OPTIONS_DISABLE_UPDATE_FEATURE      |